safe/gui/widgets/dock.py

```diff
@@ -1396,3 +1404,65 @@ def _validate_question_area(self):
             return False, message
         else:
             return True, None
+
+    def set_provenance_to_project_variables(self, provenances):
+        """Helper method to update / create provenance in project variables.
+
+        :param provenances: Keys and values from provenances.
+        :type provenances: dict
+        """
+        def write_project_variable(key, value):
+            """Helper to write project variable for base_key and value.
+
+            The key will be:
+            - base_key__KEY: value for dictionary.
+            - base_key__INDEX: value for list, tuple, set.
+            - date will be converted to ISO.
+            - None will be converted to ''.
+
+            :param key: The key.
+            :type key: basestring
+
+            :param value: A list of dictionary.
+            :type value: dict, list, tuple, set
+            """
+            if isinstance(value, (list, tuple, set)):
+                for element_index, element_value in enumerate(value):
+                    write_project_variable(
+                        '%s__%s' % (key, element_index), element_value)
+            elif isinstance(value, dict):
+                for dict_key, dict_value in value.items():
+                    write_project_variable(
+                        '%s__%s' % (key, dict_key), dict_value)
+            elif isinstance(value, (bool, str, unicode, Number)):
```

Comment:
Hmm, I have checked in the SideCI. It seems we need to set the flake8 preference.
> You can customize the behaviour and some other options by adding a `tox.ini` or `setup.cfg` file to your repository.

I will do it later (not in this PR).